packages/atlas/src/utils/time.ts

```diff
@@ -71,10 +71,11 @@ export const convertDateFormat = (timestamp: Date | string) => {
   return timestamp instanceof Date ? timestamp : parseISO(timestamp)
 }
 
-export const getNextFriday = (date = new Date()) => {
+// weekday: 1 - monday, 2 - tuesday ... 7- sunday
+export const getNextWeekday = (date = new Date(), weekday = 1) => {
   const nextDate = new Date(date)
   const currentDayOfWeek = nextDate.getDay()
-  const daysToAdd = currentDayOfWeek <= 5 ? 5 - currentDayOfWeek : 7 - currentDayOfWeek + 5
+  const daysToAdd = currentDayOfWeek <= weekday ? weekday - currentDayOfWeek : 7 - currentDayOfWeek + weekday
   nextDate.setDate(nextDate.getDate() + daysToAdd)
```

Comment:
I got confused by this...
Since `currentDayOfWeek` is `nextDate.getDay()` shouldn't it be:
```suggestion
  const daysToAdd = currentDayOfWeek <= weekday ? weekday : 7 - weekday
  nextDate.setDate(daysToAdd)
```